internal/controller/bucket/conversions.go

```diff
@@ -125,16 +125,25 @@ func LateInitialize(params *v1alpha1.BucketParameters, obs *domain.Bucket) bool
 	sort.SliceStable(obs.RetentionRules, func(i, j int) bool {
 		return string(obs.RetentionRules[i].Type) < string(obs.RetentionRules[j].Type)
 	})
-	for i := range params.RetentionRules {
-		if params.RetentionRules[i].Type == string(obs.RetentionRules[i].Type) {
-			params.RetentionRules[i].ShardGroupDurationSeconds = li.LateInitializeInt64Ptr(params.RetentionRules[i].ShardGroupDurationSeconds, obs.RetentionRules[i].ShardGroupDurationSeconds)
+
+	if len(params.RetentionRules) == 0 && len(obs.RetentionRules) > 0 {
+		for i := range params.RetentionRules {
+			{
+				if params.RetentionRules[i].Type == string(obs.RetentionRules[i].Type) {
```

Comment:
Need to check obs length here as well before accessing it.

Reply:
I see a problem with my code here :(, as it will never enter the loop in 130 since `len(params.RetentionRules)` is 0. I'm thinking of a solution.